include/rpt/compartmentalization.h

```diff
@@ -0,0 +1,156 @@
+/* ---------------------------------------------------------------------
+ *
+ * Copyright (C) 2020 - by the Lethe authors
+ *
+ * This file is part of the Lethe library
+ *
+ * The Lethe library is free software; you can use it, redistribute
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 3.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+*/
+
+#ifndef lethe_Compartmentalization_h
+#define lethe_Compartmentalization_h
+
+/**
+ * This class compartmentalize the a geometry/reactor based on physical
+ * values such as velocity, magnetic field,.... and calculate the
+ * flux between each compartment based on the velocity.
+ */
+
+#include <core/pvd_handler.h>
+#include <deal.II/base/config.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/distributed/tria.h>
+#include <map>
+#include <vector>
+#include <deal.II/base/point.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/utilities.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/lac/vector.h>
+#include <unordered_set>
+
+using namespace dealii;
+
+template <int dim>
+class Compartmentalization
+{
+public:
+  /**
+   * @brief Constructor for the Compartmentalization
+   */
+  Compartmentalization<dim>();
+
+  /**
+   * @brief Set up grid and compartmentalize first based on electric values
+   * and then find the final pattern based on velocity
+   */
+  void
+  test();
+
+private:
+  /**
+   * @brief Read the physical property (in this problem, the electric field)
+   * and write it into a vector of a vector.
+   */
+  auto
+  read_physical_property() -> std::vector<std::vector<double>>;
+
+  /**
+   * @brief Read the velocity magnitude and write it into a vector of a vector.
```

Comment:
Does it create a vector of a vector or a map?

Reply:
A map!